samples/template_ble/src/main.c

```diff
@@ -105,14 +109,47 @@ static void app_setup(void)
 	#endif
 }
 
+static enum { DFU_APPLICATION, SIDEWALK_APPLICATION } Application_to_start(){
+	settings_subsys_init();
+	settings_load();
+
+	#if defined(CONFIG_SIDEWALK_DFU_SERVICE_BLE)
+	bool DFU_mode = false;
+
+	(void) load_immediate_value(CONFIG_DFU_FLAG_SETTINGS_KEY, &DFU_mode, sizeof(DFU_mode));
+
+	if (DFU_mode) {
+		DFU_mode = false;
+		int rc = settings_save_one(CONFIG_DFU_FLAG_SETTINGS_KEY, (const void *)&DFU_mode,
+					   sizeof(DFU_mode));
+		if (rc) {
+			LOG_ERR("Failed to erase DFU flag from persistant storage, Err = %d", rc);
+		}
+		return DFU_APPLICATION;
+	}
+	#endif /* CONFIG_SIDEWALK_DFU_SERVICE_BLE */
+	return SIDEWALK_APPLICATION;
+}
+
 int main(void)
 {
 	PRINT_SIDEWALK_VERSION();
 
-	app_setup();
-	if (app_thread_init(&app_context)) {
-		LOG_ERR("Failed to start Sidewalk thread");
+	switch (Application_to_start()) {
+	case SIDEWALK_APPLICATION: {
+		app_setup();
+		if (app_thread_init(&app_context)) {
+			LOG_ERR("Failed to start Sidewalk thread");
+		}
+		break;
+	};
+	case DFU_APPLICATION: {
+		const int ret = nordic_dfu_ble_start();
+		LOG_INF("DFU service started, return value %d", ret);
+		break;
+	}
```

Comment:
Where is default? Is coverity OK with missing it?

Reply:
added default case